### PR TITLE
Adding configurable option for retry_limit on the AWS SDK, fixes #214

### LINKED ISSUE #214: AWS::EC2::Errors::RequestLimitExceeded: Request limit exceeded - when launching many instances in a batch
stack trace:
```
Generated at 2015-05-27 16:14:54 -0700
AWS::EC2::Errors::RequestLimitExceeded: machine_batch[fly_my_pretties_fly-loadtester-uswest-2a] (ec-harness::loadtesters line 86) had an error: AWS::EC2::Errors::RequestLimitExceeded: Request limit exceeded.
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/aws-sdk-v1-1.64.0/lib/aws/core/client.rb:375:in `return_or_raise'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/aws-sdk-v1-1.64.0/lib/aws/core/client.rb:476:in `client_request'
(eval):3:in `run_instances'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/aws-sdk-v1-1.64.0/lib/aws/ec2/instance_collection.rb:304:in `create'
/Users/irving/.chefdk/gem/ruby/2.1.0/gems/chef-provisioning-aws-1.2.1/lib/chef/provisioning/aws_driver/driver.rb:1005:in `block in create_many_instances'
/opt/chefdk/embedded/apps/chef/lib/chef/chef_fs/parallelizer/parallel_enumerable.rb:267:in `call'
/opt/chefdk/embedded/apps/chef/lib/chef/chef_fs/parallelizer/parallel_enumerable.rb:267:in `process_input'
/opt/chefdk/embedded/apps/chef/lib/chef/chef_fs/parallelizer/parallel_enumerable.rb:257:in `process_one'
/opt/chefdk/embedded/apps/chef/lib/chef/chef_fs/parallelizer.rb:93:in `call'
/opt/chefdk/embedded/apps/chef/lib/chef/chef_fs/parallelizer.rb:93:in `worker_loop'
```

### PR DESCRIPTION
**Pull request by tyler-ball**
\cc @jkeiser @irvingpop @randomcamel 

The README and CHANGELOG should detail these changes - let me know if it does not.

Fixes #214

### CODE REVIEW

File: lib/chef/provisioning/aws_driver/driver.rb [hunk 1]
```diff
@@ -73,13 +73,15 @@ def initialize(driver_url, config)
...
+      Chef::Config.chef_provisioning ||= {}
       ::Aws.config.update(
         credentials: credentials2.get_credentials,
         region: region || ENV["AWS_DEFAULT_REGION"] || credentials[:region],
         # TODO when we get rid of V1 replace the credentials class with something that knows how
         # to read ~/.aws/config
         :http_proxy => credentials[:proxy_uri] || nil,
-        logger: Chef::Log.logger
+        logger: Chef::Log.logger,
+        retry_limit: Chef::Config.chef_provisioning[:aws_retry_limit] || 3
```
Comment (irvingpop) [inline lib/chef/provisioning/aws_driver/driver.rb]: Hey @tyler-ball is there any reason the default retry limit shouldn't match the example in the readme?   25 seems like a sane number,  3 feels too low to be practical.

Comment (tyler-ball) [inline lib/chef/provisioning/aws_driver/driver.rb]: Sure, I can do 25.  Do you know what the default limits are from AWS before requests get throttled?

**Comment by thommay** [comment]
Should you mention that the default max is 25 requests? because it's not clear to me from the docs that that's the case

**Comment by tyler-ball** [comment]
Thanks @thommay - I added a note about the current default and dropped the default down to 5 because I think 25 is a big excessive.  I also updated the example so it shows how to set it for 10 retries instead of 25.

**Comment by thommay** [comment]
:+1:

**Pull request merged**